ui/app/src/utils/content.ts

```diff
@@ -997,6 +997,15 @@ export const openItemEditor = (
   if (type === 'form') {
     dispatch(showEditDialog({ path: item.path, authoringBase, site: siteId, onSaveSuccess }));
   } else {
-    dispatch(showCodeEditorDialog({ site: siteId, authoringBase, path: item.path, type, onSuccess: onSaveSuccess }));
+    dispatch(
+      showCodeEditorDialog({
+        site: siteId,
+        authoringBase,
+        path: item.path,
+        type,
+        mode: getEditorMode(item.mimeType),
```

Comment:
Should also fix the mode-null thing. If mode isn't provided it should fallback to text without failure.